Engineering/SecretsManagement.md

```diff
@@ -0,0 +1,121 @@
+## Secrets Management
+
+Secrets Management refers to the way in which we protect configuration settings and other sensitive data which, if 
+made public, would allow unauthorized access to resources. Examples of secrets are usernames, passwords, api keys, SAS 
+tokens etc.
+
+We should assume any repo we work on may be made public at any time and follow the appropriate procedures, even if 
+the repo is initially private.
+
+### General Approach
+
+The general approach is to keep secrets in a separate configuration file during development which is never checked in 
+to the repo. Best practice would be to add this file name to the .gitignore to prevent it being accidentally checked in.
```

Comment:
Suggestion: Use git rebase if it is not the last commit in order to remove the secret from the logs.

Reply:
Not sure how I came to be subscribed to this - sorry for 'drive-by' comment. But if a secret has been committed to git the best thing is to roll the secret rather than to try and delete it from the repository. 'git rebase' will NOT reliably remove it from the repo.

Reply:
Added link to .gitignore. 

@rdreher @edyoung This commit specifically about dealing with secrets management in code. There is a separate section in this repo on removing secrets from a git commit.